server/src/main/java/org/elasticsearch/reservedstate/service/ReservedStateErrorTaskExecutor.java

```diff
@@ -12,35 +12,33 @@
 import org.apache.logging.log4j.Logger;
 import org.elasticsearch.action.ActionResponse;
 import org.elasticsearch.cluster.ClusterState;
-import org.elasticsearch.cluster.ClusterStateTaskExecutor;
+import org.elasticsearch.cluster.SimpleBatchedExecutor;
 
 /**
  * Reserved cluster error state task executor
  * <p>
  * We use this task executor to record any errors while updating and reserving the cluster state
  */
-record ReservedStateErrorTaskExecutor() implements ClusterStateTaskExecutor<ReservedStateErrorTask> {
+class ReservedStateErrorTaskExecutor extends SimpleBatchedExecutor<ReservedStateErrorTask> {
+
     private static final Logger logger = LogManager.getLogger(ReservedStateErrorTaskExecutor.class);
 
     @Override
-    public ClusterState execute(BatchExecutionContext<ReservedStateErrorTask> batchExecutionContext) {
-        var updatedState = batchExecutionContext.initialState();
-
-        for (final var taskContext : batchExecutionContext.taskContexts()) {
-            final var task = taskContext.getTask();
-            if (task.shouldUpdate(updatedState)) {
-                try (var ignored = taskContext.captureResponseHeaders()) {
-                    updatedState = task.execute(updatedState);
-                }
-            }
-            // if the task didn't run, it still 'succeeded', it just didn't have any effect
-            taskContext.success(() -> task.listener().onResponse(ActionResponse.Empty.INSTANCE));
+    public ClusterState executeTask(ReservedStateErrorTask task, ClusterState clusterState) {
+        if (task.shouldUpdate(clusterState)) {
+            return task.execute(clusterState);
         }
-        return updatedState;
+        // if the task didn't run, it still 'succeeded', it just didn't have any effect
```

Comment:
I think this was just to explain the slightly counterintuitive `success` call, but that call is now gone so I think the comment can go too.

```suggestion
```